File: python-solid-principle-sample-code/com/principle/solid/interface_segregation/Cube.py
from Shape import Shape
from ThreeDimensionalShape import ThreeDimensionalShape

"""
    Interface Segregation Principle Achieved - Used Both separate Interface for Add-On Feature into Existing Application
"""


class Cube(Shape, ThreeDimensionalShape):

    def __init__(self, length):
        self.__length = length

    def get_length(self):
        return self.__length

    def area(self):
        # formula A = 6a2
        return 6 * self.__length ** 2

    def volume(self):
        # formula A = a3
        return self.__length ** 3

File: java-solid-principle-sample-code/src/com/principle/solid/interface_segregation/Cube.java
package com.principle.solid.interface_segregation;

public class Cube implements Shape, ThreeDimensionalShape{

	private final int edge;

    public Cube(int edge) {
        this.edge = edge;
    }

    public int getEdge() {
        return edge;
    }

    @Override
    public double area() {
//    	formula A=6a2
        return 6 * Math.pow(getEdge(), 2);
    }

	@Override
	public double volume() {
//    	formula A=a3
		return Math.pow(getEdge(), 3);
	}
}
